Sprint 1 (release v1.1) features

mdebarros opened the pull request:
Features implemented:
- https://app.zenhub.com/workspace/o/mojaloop/project/issues/89
- https://app.zenhub.com/workspace/o/mojaloop/project/issues/69

mdebarros committed:
Feature/devops k8s (#2)

* Added initial k8s & helm configs to deploy folder

* Made updates for k8 to enable network iptables config within the container

* Added kubernetes.md

* Updated circle.yml to set npm repo

mdebarros committed:
Feature/#89#69 automate the ci&cd pipeline for forensic logging sidecar (#5)

* /#89#69AutomateTheCI&CDPipelineForForensic-logging-sidecar:

Added initial config for CI/CD capabilities

* Fixed one change in the circleci config for sonarqube dir

* Integration Testing scripts have been updated for CircleCI v2.0

* Updated docker file to remove .nvmrc

* Updated correct helm set values

elnyry-sam-k reviewed (commented):
CI looks good to me. Nico, please confirm CD, I can't right now.

Commit:
Feature/#102 upgrade the version of node (#8)

6.5.0 to 8.9.4

mdebarros committed:
Fixed issue in circleci v2.0 config

Commit:
Add a review of Forensic Logging Sidecar and KMS services to use for THREAT modeling later on. (#9)

mdebarros committed:
#158 Update existing CI/CD for each repo to be compatible with K8s London (#10)

* updated gitignore to ignore intellij project files

* 1. Re-aligned circleci2.0 config to k8s-london
1.1. added support for token authorization
1.2. added support for selecting a chart version
1.3. updated helm install/update to correctly check for existing helm installation
1.4. cleaned up config
1.5. fixed bug on helm update where one MUST include the original value file to correctly merge value changes
1.6. fixed typos in integration & functional tests in config

* 2. Fixed the following issues:

2.1. Integration Test Command now does not cat, thus returns proper result code (0 - success, >0 - failure)
2.2. Integration script re-worked to properly handle failure case
2.3. updated nvmrc to the version of node that we are using

* Fixed a typo

rmothilal committed:
Feature/health check (#3)

* changes for health check

* Fixes for logging and adding port to iptables

* Neatening of code, fixing tests and fixing docker config.

mdebarros commented:
Cant even remember what this was for.
